bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java

```diff
@@ -436,6 +443,86 @@ private TableInfo materializeTable(
     }
   }
 
+  public void loadDataIntoTable(
+      LoadDataOptions options,
+      List<String> sourceUris,
+      FormatOptions formatOptions,
+      JobInfo.WriteDisposition writeDisposition) {
+    Job finishedJob = null;
+    try {
+      LoadJobConfiguration.Builder jobConfiguration =
+          LoadJobConfiguration.newBuilder(options.getTableId(), sourceUris, formatOptions)
+              .setCreateDisposition(JobInfo.CreateDisposition.CREATE_IF_NEEDED)
+              .setWriteDisposition(writeDisposition)
+              .setAutodetect(true);
+
+      options.getCreateDisposition().ifPresent(jobConfiguration::setCreateDisposition);
+
+      if (options.getPartitionField().isPresent() || options.getPartitionType().isPresent()) {
+        TimePartitioning.Builder timePartitionBuilder =
+            TimePartitioning.newBuilder(options.getPartitionTypeOrDefault());
+        options.getPartitionExpirationMs().ifPresent(timePartitionBuilder::setExpirationMs);
+        options
+            .getPartitionRequireFilter()
+            .ifPresent(timePartitionBuilder::setRequirePartitionFilter);
+        options.getPartitionField().ifPresent(timePartitionBuilder::setField);
+        jobConfiguration.setTimePartitioning(timePartitionBuilder.build());
+        options
+            .getClusteredFields()
+            .ifPresent(
+                clusteredFields -> {
+                  Clustering clustering =
+                      Clustering.newBuilder().setFields(clusteredFields).build();
+                  jobConfiguration.setClustering(clustering);
+                });
+      }
+
```

Comment:
In the v1 implementation, we are also using the option `options.isUseAvroLogicalTypes`. Are we not using it here?

Reply:
Good catch!